### PR TITLE
Cloudwatch: Deprecate cloudwatchNewRegionsHandler feature toggle and remove core imports from featuremgmt

### LINKED ISSUE #80899: Remove core imports from `grafana/pkg/services/featuremgmt` in CloudWatch

### PR DESCRIPTION
**Pull request by njvrzm**
**What is this feature?**

This removes some core grafana imports from Cloudwatch. Also removes the `cloudwatchNewRegionsHandler` feature, which is obsolete.

**Why do we need this feature?**

This is part of decoupling cloudwatch from core grafana, #78778.

**Which issue(s) does this PR fix?**:
Fixes: #80899

Please check that:
- [ ] It works as expected from a user's perspective.

### CODE REVIEW

**Review by iwysiu — APPROVED** [review]
A couple clean up comments, but nice work!
I tried shutting off the cross account querying flag, and it did successfully use the old log group picker.
I would double check (maybe in slack?) that we're ok to remove that flag before we remove it and that none of the users are having issues.

File: pkg/tsdb/cloudwatch/features/features.go [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+)
+
+const (
+	FlagCloudWatchCrossAccountQuerying    = "cloudWatchCrossAccountQuerying"
+	FlagCloudWatchWildCardDimensionValues = "cloudWatchWildCardDimensionValues"
+	FlagCloudWatchBatchQueries            = "cloudWatchBatchQueries"
+)
+
+func IsEnabled(ctx context.Context, feature string) bool {
```
Comment (sarahzinger) [inline pkg/tsdb/cloudwatch/features/features.go:15]: this is really neat, does this mean we could have backend feature toggles in our external datasource plugins if we implemented something similar there? 👀 very nifty!

**Review by sarahzinger — COMMENTED** [review]
Looking good! Lets just make sure the leftover comment bits are removed first!

**Comment by sarahzinger** [comment]
oh also random, but I tend to mark _nearly_ everything in Add To Changelog. In this case in particular where we are deprecating a feature toggle I think it would be good. We haven't seen bugs come in to my knowledge, but you never know who might have turned off the feature toggle on their own on prem instance or something like that without reporting a bug. This way if something unexpected happens with Cloudwatch after an upgrade a user can quickly scan the changelog and see some possible threads to pull on to try to figure out what happened.

To that end I might change the name of this pr from `Cloudwatch: remove core imports from featuremgmt` to `Cloudwatch: Deprecate cloudwatchNewRegionsHandler feature toggle and remove core imports from featuremgmt`

File: pkg/tsdb/cloudwatch/test_utils.go [hunk 2]
```diff
@@ -270,3 +272,9 @@ func (e fakeAWSError) Code() string {
 func (e fakeAWSError) Message() string {
 	return e.message
 }
+
+func ContextWithFeaturesEnabled(enabled ...string) context.Context {
```
[the remark this answers is not in the document]
Comment (njvrzm) [reply pkg/tsdb/cloudwatch/test_utils.go:276]: Indeed! I thought it did at some point - maybe I was using it in a subpackage? - but it sure doesn't now

**Comment by njvrzm** [comment]
Re: removing the cross account querying flag, I got the 👍🏻 from @sarahzinger in slack for that.

**Pull request merged**